automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -167,6 +167,39 @@ trait LeonardoTestUtils
     (seen - "clusterServiceAccount" - "notebookServiceAccount") shouldBe (expected - "clusterServiceAccount" - "notebookServiceAccount")
   }
 
+  def gceLabelCheck(seen: LabelMap,
+                    runtimeName: RuntimeName,
+                    googleProject: GoogleProject,
+                    creator: WorkbenchEmail,
+                    runtimeRequest: RuntimeRequest): Unit = {
+
+    // the SAs can vary here depending on which ServiceAccountProvider is used
+    // set dummy values here and then remove them from the comparison
+    // TODO: check for these values after tests are agnostic to ServiceAccountProvider ?
+
+    val dummyClusterSa = WorkbenchEmail("dummy-cluster")
+    val dummyNotebookSa = WorkbenchEmail("dummy-notebook")
+    val jupyterExtensions = runtimeRequest.userJupyterExtensionConfig match {
+      case Some(x) => x.nbExtensions ++ x.combinedExtensions ++ x.serverExtensions ++ x.labExtensions
+      case None    => Map()
+    }
+
+    val expected = runtimeRequest.labels ++ DefaultLabelsCopy(
+      runtimeName,
+      googleProject,
+      creator,
+      Some(dummyClusterSa),
+      Some(dummyNotebookSa),
+      runtimeRequest.jupyterExtensionUri,
+      runtimeRequest.jupyterUserScriptUri,
+      runtimeRequest.jupyterStartUserScriptUri,
+      runtimeRequest.toolDockerImage.map(getExpectedToolLabel).getOrElse("Jupyter")
+    ).toMap ++ jupyterExtensions
+
+    (seen - "clusterServiceAccount" - "notebookServiceAccount") shouldBe (expected - "clusterServiceAccount" - "notebookServiceAccount")
```

Comment:
I wonder if we can remove some of labels being checked...Talked with Isaac about labels a bit and I think some of ones that duplicate with top level fields can be removed (such as runtimeName, googleProject, creator)